gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -93,14 +115,18 @@ private GrpcCallContext(
       @Nullable Duration streamWaitTimeout,
       @Nullable Duration streamIdleTimeout,
       @Nullable Integer channelAffinity,
-      ImmutableMap<String, List<String>> extraHeaders) {
+      ImmutableMap<String, List<String>> extraHeaders,
+      @Nullable RetrySettings retrySettings,
+      @Nullable Set<StatusCode.Code> retryableCodes) {
     this.channel = channel;
     this.callOptions = Preconditions.checkNotNull(callOptions);
     this.timeout = timeout;
     this.streamWaitTimeout = streamWaitTimeout;
     this.streamIdleTimeout = streamIdleTimeout;
     this.channelAffinity = channelAffinity;
     this.extraHeaders = Preconditions.checkNotNull(extraHeaders);
+    this.retrySettings = retrySettings;
+    this.retryableCodes = retryableCodes == null ? null : ImmutableSet.copyOf(retryableCodes);
```

Comment:
Why is this nullable? per Effective Java, empty collections are preferred.

Reply:
There is a difference in this case:
* `null` means 'do not override the default settings`.
* An empty set means 'override the default settings by disabling retrying'.

I've added that to the documentation of `ApiCallContext#getRetryableCodes()`.